File: ec_production/ec_production/report/payment_report/payment_report.py
# Copyright (c) 2026, SkylineBiz Private Limited and contributors
# For license information, please see license.txt

import frappe
from frappe import _


def execute(filters=None):
    filters = filters or {}

    columns = get_columns()
    data = get_data(filters)

    return columns, data


def get_columns():
    return [
        {
            "label": _("Employee"),
            "fieldname": "employee",
            "fieldtype": "Link",
            "options": "Employee",
            "width": 250,
        },
        {
            "label": _("Final Amount"),
            "fieldname": "final_amount",
            "fieldtype": "Currency",
            "width": 150,
        },
    ]


def get_data(filters):
    from_date = filters.get("from_date")
    to_date = filters.get("to_date")

    process_lots = frappe.get_all(
        "EC Process Lot",
        filters={
            "date": ["between", [from_date, to_date]]
        },
        fields=["name"]
    )

    if not process_lots:
        return []

    lot_names = [d.name for d in process_lots]

    lot_items = frappe.get_all(
        "EC Process Lot Item",
        filters={
            "parent": ["in", lot_names]
        },
        fields=[
            "employee",
            "amount"
        ]
    )
    
    employee_names = {
		e.name: e.employee_name
		for e in frappe.get_all(
			"Employee",
			fields=["name", "employee_name"]
		)
	}

    employee_totals = {}

    for row in lot_items:
        if not row.employee:
            continue

        employee_totals.setdefault(row.employee, 0)
        employee_totals[row.employee] += row.amount or 0

    data = []

    for employee, total in sorted(employee_totals.items()):
        data.append({
            "employee": employee_names.get(employee, employee),
            "final_amount": total
        })

    return data

File: ec_production/ec_production/report/payment_report/payment_report.js
// Copyright (c) 2026, SkylineBiz Private Limited and contributors
// For license information, please see license.txt

frappe.query_reports["Payment Report"] = {
    filters: [
        {
            fieldname: "from_date",
            label: __("From"),
            fieldtype: "Date",
            reqd: 1,
            default: frappe.datetime.month_start()
        },
        {
            fieldname: "to_date",
            label: __("To"),
            fieldtype: "Date",
            reqd: 1,
            default: frappe.datetime.month_end()
        }
    ],

    formatter(value, row, column, data, default_formatter) {
        value = default_formatter(value, row, column, data);

        if (column.fieldname === "employee" && data) {
            return `
                <a href="javascript:void(0)"
                onclick="open_voucher_report(
                        '${data.employee_id}',
                        '${frappe.query_report.get_filter_value('from_date')}',
                        '${frappe.query_report.get_filter_value('to_date')}'
                )">
                    ${value}
                </a>
            `;
        }

        return value;
    }
};

window.open_voucher_report = function (employee, from_date, to_date) {

    frappe.route_options = {
        employee: employee,
        from_date: from_date,
        to_date: to_date
    };

    frappe.set_route("query-report", "Voucher Report");
};